### PR TITLE
Do session/renew when update_leader is called

### PR DESCRIPTION
**Pull request by CyberDem0n**
should fix https://github.com/zalando/patroni/issues/329

in addition to that make it possible to specify ttl in Consul smaller than 20 (10 seconds).

### CODE REVIEW

File: features/environment.py [hunk 1]
```diff
@@ -192,9 +192,9 @@ def __init__(self, context, mktemp=True):
     def _is_accessible(self):
         return self._is_running()
 
-    def stop_and_remove_work_directory(self, timeout=15):
+    def stop(self, kill=False, timeout=15):
```
Comment (alexeyklyukin) [inline features/environment.py]: I wonder if we should also pass kill to the AbastractController.

**Comment by alexeyklyukin** [comment]
👍

**Comment by alexeyklyukin** [comment]
👍

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**